src/main/java/apoc/merge/Merge.java

```diff
@@ -22,49 +23,88 @@
             throw new IllegalArgumentException("you need to supply at least one identifying property for a merge");
         }
 
-        String labels = labelNames.stream().map(s -> wrapInBacktics(s)).collect(Collectors.joining(":"));
+        String labels = labelNames.stream().map(s -> Util.quote(s)).collect(Collectors.joining(":"));
 
         Map<String, Object> params = new HashMap<>();
-        params.put("props", Util.merge(identProps, props));
+        params.put("identProps", identProps);
+        params.put("props", props);
         String identPropsString = buildIdentPropsString(identProps);
 
         final String cypher = "MERGE (n:" + labels + "{" + identPropsString + "}) ON CREATE SET n += $props RETURN n";
         Node node = Iterators.single(db.execute(cypher, params ).columnAs("n"));
         return Stream.of(new NodeResult(node));
     }
 
-    private String wrapInBacktics(String s) {
-        return "`" + s + "`";
+    @Procedure(value="apoc.merge.node.eager", mode = Mode.WRITE, eager = true)
+    @Description("apoc.merge.node.eager(['Label'], identProps:{key:value, ...}, config:{onCreateProps:{key:value,...}, onMatchProps:{key:value,...}}) - merge node with dynamic labels, with support for setting properties ON CREATE or ON MATCH")
+    public Stream<NodeResult> nodeEager(@Name("label") List<String> labelNames, @Name("identProps") Map<String, Object> identProps, @Name(value="config", defaultValue = "{}") Map<String, Object> config) {
```

Comment:
I wouldn't put the extra props into config but into optional parameters which default to empty maps.